cmake/kokkos_functions.cmake

```diff
@@ -698,3 +698,73 @@ FUNCTION(kokkos_link_tpl TARGET)
   ENDIF()
 ENDFUNCTION()
 
+FUNCTION(COMPILER_SPECIFIC_OPTIONS_HELPER)
+  SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang GNU)
+  CMAKE_PARSE_ARGUMENTS(
+    PARSE
+    "LINK_OPTIONS;COMPILE_OPTIONS;COMPILE_DEFINITIONS"
+    ""
+    "${COMPILERS}"
+    ${ARGN})
+
+  SET(COMPILER ${KOKKOS_CXX_COMPILER_ID})
+
+  SET(FLAGS)
```

Comment:
"FLAGS" is not a great variable name because it is recognized as an argument separator keyword and highlighted.

Reply:
Yeah, that was lazy of me. Touche.

Reply:
renamed to COMPILER_SPECIFIC_FLAGS_TMP (we already got a function COMPILER_SPECIFIC_FLAGS